File: backend/app/schemas/family_member.py
from datetime import datetime

from pydantic import BaseModel


class FamilyMemberCreate(BaseModel):
    name: str
    color: str = "#0052CC"
    avatar_emoji: str = "👤"


class FamilyMemberUpdate(BaseModel):
    name: str | None = None
    color: str | None = None
    avatar_emoji: str | None = None


class FamilyMemberResponse(BaseModel):
    id: int
    name: str
    color: str
    avatar_emoji: str
    created_at: datetime

    model_config = {"from_attributes": True}

File: ios/Familienkalender/Models/FamilyMember.swift
import Foundation

struct FamilyMemberCreate: Codable {
    let name: String
    var color: String = "#0052CC"
    var avatarEmoji: String = "👤"
}

struct FamilyMemberUpdate: Codable {
    var name: String?
    var color: String?
    var avatarEmoji: String?
}

struct FamilyMemberResponse: Codable, Identifiable, Hashable {
    let id: Int
    let name: String
    let color: String
    let avatarEmoji: String
    let createdAt: String
}
